backend/src/main/kotlin/com/ci5644/trade/services/auth/AuthorizationService.kt

```diff
@@ -0,0 +1,103 @@
+package com.ci5644.trade.services.auth
+
+import com.ci5644.trade.config.SecurityConstants
+import com.ci5644.trade.config.JWT.JWTSecurityUtils
+import com.ci5644.trade.dto.auth.RegisterDTO
+import org.springframework.security.core.userdetails.UsernameNotFoundException
+import com.ci5644.trade.exceptions.runtime.UsernameTakenException
+import com.ci5644.trade.models.user.UserEntity
+import com.ci5644.trade.repositories.UserRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpCookie
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Service
+import javax.naming.AuthenticationException
+import org.springframework.security.crypto.password.PasswordEncoder
+
+/**
+ * Service class for handling authorization-related operations.
+ */
+@Service
+class AuthorizationService {
+
+    @Autowired
+    private lateinit var authenticationManager: AuthenticationManager
```

Comment:
Autowired is an old practice only reserved for special cases, in this case please add them to the constructor